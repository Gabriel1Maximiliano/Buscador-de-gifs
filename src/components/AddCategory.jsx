import React, {  useState } from 'react'
import PropTypes from 'prop-types'


export const AddCategory = ({handleAddCatgories}) => {
    const [value, setValue] = useState('');

    const handleInputChange = ({target}) => {
      setValue(target.value);
    }
    
    const handleSubmit =(e) => {
  let reGex= /[a-zäÄëËïÏöÖüÜáéíóúáéíóúÁÉÍÓÚÂÊÎÔÛâêîôûàèìòùÀÈÌÒÙ0-9]/
        if(value.trim().length <= 1) return alert('Inserte un nombre');
        if(!reGex.test(value)) return alert('Caracteres inválidos')
        e.preventDefault();
        handleAddCatgories(value)
       // setCategories((categories)=>[...categories,value])
        setValue('')
    }
  return (
   <form onSubmit={handleSubmit} aria-label='form' >
        <input
        type='text'
        placeholder='buscar gifs'
        value={value}
        onChange={handleInputChange}
        />
    </form>
  )
}

AddCategory.propTypes ={
  handleAddCatgories: PropTypes.func.isRequired
}