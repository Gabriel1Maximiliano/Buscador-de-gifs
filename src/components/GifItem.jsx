import React from 'react'
import PropTypes from 'prop-types'
export const GifItem = ({id,title,img,handleEliminate}) => {
   
  return (
    <div className='card'>
        <img src={ img } alt='imagen'></img>
        <p>{title}</p>
        <button onClick={(e)=>handleEliminate(id)}>Eliminar</button>
    </div>
  )
}

 GifItem.propTypes = {
   title: PropTypes.string.isRequired,
   img  : PropTypes.string.isRequired,
 }