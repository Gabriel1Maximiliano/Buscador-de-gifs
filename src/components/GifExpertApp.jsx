import React, { useState } from 'react'
import { AddCategory } from './AddCategory';
import { GifGrid } from './GifGrid';

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['One Punch']);
 
     const handleAddCatgories = (e) => {
    if(categories.includes(e)){
        return;
    }else{

        setCategories(
           [e,...categories]
          )
    }
        
     }
   
  return (
    <>
    <h1>GifExpertApp</h1>
    <AddCategory  
    handleAddCatgories={ handleAddCatgories } 
    />
   
    
   {
    categories.map((c)=>{
        return (
            <GifGrid key={c}  category={c}/>
        )
    })
   }
   
    </>
  )
}
