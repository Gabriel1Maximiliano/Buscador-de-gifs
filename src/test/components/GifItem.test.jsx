import { render,screen } from "@testing-library/react"

 import{GifItem} from '../../components/GifItem'
 
 describe('primer Prueba sobre < gifItem />', () => {
    test('probando el metodo container', () => {

        const title ='hola';
        
        const img   ='https://media3.giphy.com/media/GRSnxyhJnPsaQy9YLn/giphy.gif?cid=c6f82f11cvpt7ikruu8ygixmporcfncou0su74bp9llp2u3k&amp;rid=giphy.gif&amp;ct=g';

        const alt="imagen"
        
        const {container} = render(<GifItem title={title} img={img} /> );
      
    })
 })

 describe('segunda prueba sobre <GifItem ></GifItem>', () => {
    test('probando el snapshot', () => {

        const title ='hola';
       
        const img   ='https://media3.giphy.com/media/GRSnxyhJnPsaQy9YLn/giphy.gif?cid=c6f82f11cvpt7ikruu8ygixmporcfncou0su74bp9llp2u3k&amp;rid=giphy.gif&amp;ct=g';

        const alt="imagen"
        const {container} = render(<GifItem title={title} img={img} /> );
        expect( container ).toMatchSnapshot();
        
    })
 })

 describe('probando el render con debug sobre <Gifitem></Gifitem>', () => {

    test('debe usar debug para ver componente', () => {

        const title ='hola';

        const img   ='https://media3.giphy.com/media/GRSnxyhJnPsaQy9YLn/giphy.gif?cid=c6f82f11cvpt7ikruu8ygixmporcfncou0su74bp9llp2u3k&amp;rid=giphy.gif&amp;ct=g';

        const alt="imagen";

      render(<GifItem title={title} img={img}/>);

    })

 })

 describe('Probando el screen', () => {
    test('debe mostrar el valor de URL y ALT de la imagen ', () => {
        const title ='hola';

        const img   ='https://media3.giphy.com/media/GRSnxyhJnPsaQy9YLn/giphy.gif?cid=c6f82f11cvpt7ikruu8ygixmporcfncou0su74bp9llp2u3k&amp;rid=giphy.gif&amp;ct=g';

        const altl="imagen";




        const componente = render(<GifItem title={title} img={img}/>);

        const {src,alt} = screen.getByRole('img');
        expect(src).toBe( img );
        expect(altl).toBe( alt );


    })

    test('Debe mostrar el tÃ­tulo', () => {
        const title ='hola';

        const img   ='https://media3.giphy.com/media/GRSnxyhJnPsaQy9YLn/giphy.gif?cid=c6f82f11cvpt7ikruu8ygixmporcfncou0su74bp9llp2u3k&amp;rid=giphy.gif&amp;ct=g';


        render(<GifItem title={title} img={img}/>);

        expect( screen.getByText( title )).toBeTruthy() 

   

    })

 })