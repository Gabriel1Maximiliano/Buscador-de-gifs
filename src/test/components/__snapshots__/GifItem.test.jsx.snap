// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`segunda prueba sobre <GifItem ></GifItem> probando el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="imagen"
      src="https://media3.giphy.com/media/GRSnxyhJnPsaQy9YLn/giphy.gif?cid=c6f82f11cvpt7ikruu8ygixmporcfncou0su74bp9llp2u3k&amp;rid=giphy.gif&amp;ct=g"
    />
    <p>
      hola
    </p>
    <button>
      Eliminar
    </button>
  </div>
</div>
`;
